eclipse-collections/src/main/java/org/eclipse/collections/impl/list/primitive/IntInterval.java

```diff
@@ -1007,4 +1016,33 @@ public int next()
             throw new NoSuchElementException();
         }
     }
+
+    private class IntIntervalSpliterator implements Spliterator.OfInt {
+        AtomicInteger current = new AtomicInteger(from);
+
+        @Override
+        public OfInt trySplit()
+        {
+            return null;
+        }
+
+        @Override
+        public long estimateSize()
+        {
+            return size();
+        }
+
+        @Override
+        public int characteristics()
+        {
+            return Spliterator.DISTINCT|Spliterator.IMMUTABLE|Spliterator.NONNULL|Spliterator.ORDERED|Spliterator.SIZED|Spliterator.SORTED;
+        }
+
+        @Override
+        public boolean tryAdvance(IntConsumer action)
+        {
+            action.accept(current.getAndAdd(step));
+            return current.get() < to;
+        }
+    }
```

Comment:
In the same way that `Spliterators` (the factory class) can wrap Iterators and IntIterators, we might consider a few forms of extra compatibility.

We could write adaptors that take our primitive iterators and turn them into spliterators.
We could make 3 of our primitive iterators extend the JDK primitive iterators.
We could make 3 of our primitive iterables extend the JDK primitive iterables.

Interval is one of the few iterables that's splittable at any point. It would be great to implement trySplit() for real.

Ideally IntervalSpliterator wouldn't need the outer reference. That is, it would be either a static inner class or a top-level class.

Reply:
Sure, that's a great approach. I'd still recommend replacing AtomicInteger with int. And without code sharing with the iterator, we'd just need to ensure good test coverage.

Reply:
I don't see any javadoc anywhere that says anything about `tryAdvance` or `trySplit` thread safety, so I assume they don't have to be (the JDK implementations aren't). So I think it's good to replace AtomicInteger with int.